CMP1903M Assessment 1 Base Code/Program.cs

```diff
@@ -1,63 +1,60 @@
-                    testString = ioReader.manualTextInput();
-                }
-                else if (usrInput == "2")
-                {
-                    Console.WriteLine("Enter absolute file path");
-                    testString = ioReader.fileTextInput(Console.ReadLine());
-                }
-                else if (usrInput == "3") break;
-                else
-                {
-                    Console.WriteLine("Invalid input!");
-                    continue;
-                }
-                
-                //Create an 'Analyse' object
-                //Pass the text input to the 'analyseText' method
-                //Receive a list of integers back
-                var dict = Analyse.AnalyseText(testString);
-
-                //Report the results of the analysis
-                Console.WriteLine("\n");
-                foreach (var key in dict.Keys)
-                {
-                    Console.WriteLine("{0}:   \t{1}", key, dict[key]);
-                }
-
-                Console.WriteLine("\n");
-                
-                //TO ADD: Get the frequency of individual letters?
-                
-            }
-        }
-    }
-}
+﻿//Base code project for CMP1903M Assessment 1
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+
+namespace CMP1903M_Assessment_1_Base_Code
+{
+    /// <summary>
+    /// Main program class
+    /// </summary>
+    class Program
+    {
+        /// <summary>
+        /// Entry method
+        /// </summary>
+        static void Main()
+        {
+            // Initialize IO reader
+            Input ioReader = new Input();
+            string analysisText;
+            
+            // Loop until exit
+            while (true)
+            {
+                Console.WriteLine("Options:\n1: Manual input\n2: File input\n3: Exit");
+                string usrInput = Console.ReadLine() ?? string.Empty;
+                switch (usrInput)
+                {
+                    case "1": // CLI text entry
+                        analysisText = ioReader.manualTextInput();
+                        break;
+                    case "2": // Read file
+                        analysisText = ioReader.fileTextInput();
+                        break;
+                    case "3": // Exit
+                        return;
+                    default:
+                        Console.WriteLine("Invalid input!");
+                        continue;
+                }
+                
+                // Initialise analyse object with analysisText
+                var analysis = new Analyse
+                {
+                    AnalisisText = analysisText
+                };
+
+                Console.WriteLine("\n");
+                
+                // Create report object and output to console
+                var output = new Report(in analysis);
```

Comment:
I'm not well-versed enough in the `in` keyword to identify if it's necessary, but perhaps you could just pass the analysis into the constructor method normally?
Also, try to use explicit types where you can.
```suggestion
                Report output = new Report(analysis);
```

Reply:
This is true. Although the `in` keyword provides additional functionality in this case as it allows the class instance to be reused after `analysis` changes as we are not passing the actual `analysis` object, just the reference to it in memory. However, in this case, `analysis` does not change and the `Report` instance is not reused so this could be seen as unnecessary.